R/geom_parallel_slopes.R

```diff
@@ -0,0 +1,168 @@
+#' Parallel slopes model
+#' 
+#' \code{geom_parallel_slopes()} fits parallel slopes model and adds its line
+#' output(s) to a \code{ggplot} object. Basically, it fits a unified model with
+#' intercepts varying between groups (which should be supplied as standard
+#' {ggplot2} grouping aesthetics: \code{group}, \code{color}, \code{fill},
+#' etc.).
+#'
+#' @param se Display confidence interval around model lines? \code{TRUE} by
+#'   default.
+#' @param formula Formula to use per group in parallel slopes model. Basic
+#'   linear \code{y ~ x} by default.
+#' @param n Number of points per group at which to evaluate model.
+#' @inheritParams ggplot2::layer
+#' @inheritParams ggplot2::geom_smooth
+#'
+#' @examples
+#' example_df <- house_prices %>%
+#'   dplyr::slice(1:1000) %>% 
+#'   dplyr::mutate(
+#'     log10_price = log10(price),
+#'     log10_size = log10(sqft_living)
+#'   )
+#' ggplot_viz <- ggplot2::ggplot(
+#'   example_df,
+#'   ggplot2::aes(x = log10_size, y = log10_price, color = condition)
+#' ) +
```

Comment:
After we merge we'll likely polish these a bit to make them more in line with the coding style we've been presenting to students in our intro courses. For example, we don't use `dplyr::` but rather load the libraries